numpy/core/src/umath/ufunc_object.c

```diff
@@ -2764,9 +2764,12 @@ reducelike_promote_and_resolve(PyUFuncObject *ufunc,
      * The first operand and output should be the same array, so they should
      * be identical.  The second argument can be different for reductions,
      * but is checked to be identical for accumulate and reduceat.
+     * Ideally, the type-resolver ensures that all are identical, but we do
+     * not enforce this here strictly.
```

Comment:
I would add "strictly, since on windows this seems not to be guaranteed; see gh-20699"